internal/validitywindow/handler_test.go

```diff
@@ -116,19 +119,19 @@ func TestBlockFetcherHandler_FetchBlocks(t *testing.T) {
 	}
 }
 
-func generateBlockChain(n int, containersPerBlock int) map[uint64]ExecutionBlock[container] {
+func generateBlockChain(n int, containersPerBlock int, r *rand.Rand) map[uint64]ExecutionBlock[container] {
 	genesis := newExecutionBlock(0, 0, []int64{})
 	blks := make(map[uint64]ExecutionBlock[container])
 	blks[0] = genesis
 
 	for i := 1; i < n; i++ {
 		containers := make([]int64, containersPerBlock)
 		for j := 0; j < containersPerBlock; j++ {
-			containers[j] = rand.Int63n(int64(n)) //nolint:gosec
+			containers[j] = r.Int63n(int64(n))
```

Comment:
Rather than use `r` here, can we replace the use of randomness with a deterministic function?

Reply:
Also - isn't this generating an invalid sequence of blocks since the containers may have an expiry below the execution time of the block we are including them in?